joblib/numpy_pickle.py

```diff
@@ -263,7 +264,21 @@ def save(self, obj):
 
             # The array wrapper is pickled instead of the real array.
             wrapper = self._create_array_wrapper(obj)
-            Pickler.save(self, wrapper)
+
+            if self.protocol >= 4:
+                # A framer was introduced with pickle protocol 4. Here we want
+                # to ensure the wrapper object is written before the numpy
+                # array buffer and in a dedicated frame.
+
+                # closing and force writing any opened frame.
+                self.framer.end_framing()
```

Comment:
commit_frame let the current_frame opened, so the array buffer is written within the same frame. I fear it may break things later.